src/services/unlockCourtCase.ts

```diff
@@ -1,19 +1,29 @@
 import { DataSource, UpdateResult } from "typeorm/"
+import { QueryDeepPartialEntity } from "typeorm/query-builder/QueryPartialEntity"
 import CourtCase from "./entities/CourtCase"
+import User from "./entities/User"
 
-const unlockCourtCase = async (dataSource: DataSource, courtCaseId: number): Promise<UpdateResult | Error> => {
+const unlockCourtCase = async (
+  dataSource: DataSource,
+  courtCaseId: number,
+  user: User
+): Promise<UpdateResult | Error> => {
   const courtCaseRepository = dataSource.getRepository(CourtCase)
+  const setFields: QueryDeepPartialEntity<CourtCase> = {}
 
   try {
-    return await courtCaseRepository
-      .createQueryBuilder()
-      .update(CourtCase)
-      .set({
-        errorLockedByUsername: null,
-        triggerLockedByUsername: null
-      })
-      .where("error_id = :id", { id: courtCaseId })
-      .execute()
+    const query = courtCaseRepository.createQueryBuilder().update(CourtCase)
+
+    if (user.canLockExceptions) {
+      setFields.errorLockedByUsername = null
+    }
+    if (user.canLockTriggers) {
+      setFields.triggerLockedByUsername = null
+    }
+
+    query.set(setFields).where("error_id = :id", { id: courtCaseId })
+
+    return await query.execute()
```

Comment:
I don't think we need the await when you're returning it

Reply:
Good shout! I also remove the try/catch and use `catch()` function instead.